hub-saml/src/main/java/uk/gov/ida/saml/core/validators/assertion/IdentityProviderAssertionValidator.java

```diff
@@ -25,6 +28,21 @@ public IdentityProviderAssertionValidator(
         this.subjectConfirmationValidator = subjectConfirmationValidator;
     }
 
+    public void validateConsistency(Assertion authnStatementAssertion, Assertion matchingDatasetAssertion) {
+        validateConsistency(Arrays.asList(authnStatementAssertion, matchingDatasetAssertion));
+    }
+
+    public void validateConsistency(List<Assertion> assertions) {
+        if (assertions.size() == 0) return;
+
+        String pid = assertions.get(0).getSubject().getNameID().getValue();
+        boolean pidsAreConsistent = assertions.stream().allMatch(assertion -> assertion.getSubject().getNameID().getValue().equals(pid));
```

Comment:
Would:
```
assertions.stream()
  .map(ass -> ass.getSubject().getNameID().getValue())
  .distinct()
  .count() == 1
```
be nicer do you think?

Reply:
Or you could do:

```
Integer size = assertions.size();
switch(size) {
  case 0: return;
  case 1: return;
  case 2: validateConsistency(assertions.get(0), assertions.get(1); return;
  default: throw new SamlTransformationErrorException("Unexpected number of assertions. Expected 0, 1 or 2, got " + size);
```

Reply:
Hmm, stream one probably seems most elegant.